### PR TITLE
Fixes for amg, minife, and pennant build with geopm hints

### LINKED ISSUE #2298: GEOPM App Integration build scripts fail for apps with GEOPM markup added
**Describe the bug**
I tried to build AMG, PENNANT, and MiniFE and I expected the existing build.sh script to function instead the patches adding GEOPM region markup caused build failures

**GEOPM version**
1.1.0+dev2023g90214fa2

**Expected behavior**
./build.sh should build the workloads in the integration/apps directory without error

**Actual behavior**/**Error messages produced**
```
mpiicc -o timing.o -c -I.. -O2 -DTIMER_USE_MPI -DHYPRE_USING_OPENMP -fopenmp -DHYPRE_HOPSCOTCH -DHYPRE_USING_PERSISTENT_COMM -DHYPRE_BIGINT -xCORE-AVX2 -I/home/daniel/build/geopm/include -DGEOPM timing.c timing.c(166): error: identifier "GEOPM_REGION_HINT_UNKNOWN" is undefined
geopm_prof_region(name, GEOPM_REGION_HINT_UNKNOWN, g_geopm_rid + time_index); ^
compilation aborted for timing.c (code 2)
```

**Additional context**
An additional reason to implement #2293

### PR DESCRIPTION
**Pull request by lhlawson**
Signed-off-by: Lowren Lawson <lowren.h.lawson@intel.com>

- Relates to #2298 from github issues
- Partially Fixes #2298 change request from github issues.

Updates & additions to the build commits for AMG, MiniFE, and PENNANT that support GEOPM makeup.  Tested on mcfly.  
Note that this does not fix all workloads with GEOPM markup, only those mentioned above.

### CODE REVIEW

File: integration/apps/minife/0004-Add-Epoch-markup.patch [hunk 1]
```diff
@@ -1,42 +1,43 @@
-From 3ebeb9799662a3414e74aff5e0e95884e83945ee Mon Sep 17 00:00:00 2001
+From 03b8ad88c1c8d02f472bb535e044d366654e2abf Mon Sep 17 00:00:00 2001
 From: Brad Geltz <brad.geltz@intel.com>
-Date: Wed, 7 Oct 2020 15:54:58 -0700
-Subject: [PATCH 4/5] Add Epoch markup
+Date: Wed, 7 Oct 2020 15:50:07 -0700
+Subject: [PATCH 3/5] Link GEOPM
```
Comment (bgeltz) [inline integration/apps/minife/0004-Add-Epoch-markup.patch]: Please rename this back to "Add Epoch markup".  There is no linking happening in this patch.

Also, something seems wrong because this is patch 0004-Add...patch but the "Subject" has it listed as patch 3.

File: integration/apps/minife/0005-Add-region-markup.patch [hunk 2]
```diff
@@ -1,37 +1,74 @@
-From 1aa92708e95ffb39815759f588d0371c508613c7 Mon Sep 17 00:00:00 2001
+From 03b8ad88c1c8d02f472bb535e044d366654e2abf Mon Sep 17 00:00:00 2001
 From: Brad Geltz <brad.geltz@intel.com>
-Date: Wed, 7 Oct 2020 15:55:14 -0700
-Subject: [PATCH 5/5] Add region markup
+Date: Wed, 7 Oct 2020 15:50:07 -0700
+Subject: [PATCH 3/5] Link GEOPM
```
Comment (bgeltz) [inline integration/apps/minife/0005-Add-region-markup.patch]: Please rename this back to "Add region markup".  There is no linking happening in this patch.

Also, this is patch 5 according to the file name, but the "Subject" shows it as patch 3.  There are two patch 3's right now.

File: integration/apps/minife/0005-Add-region-markup.patch [hunk 3]
```diff
@@ -105,20 +152,32 @@ index cdf9b15..4f54069 100644
...
+ 
+ template<typename VectorType>
+ void
+-  daxpby(const MINIFE_SCALAR alpha, 
++  daxpby(const MINIFE_SCALAR alpha,
+ 	const VectorType& x,
+-	const MINIFE_SCALAR beta, 
++	const MINIFE_SCALAR beta,
+ 	VectorType& y)
+ {
```
Comment (bgeltz) [inline integration/apps/minife/0005-Add-region-markup.patch]: What is this change about?

**Review by bgeltz — CHANGES_REQUESTED** [review]

File: integration/apps/minife/0005-Add-region-markup.patch (hunk 3 above)
Comment (lhlawson) [reply integration/apps/minife/0005-Add-region-markup.patch]: Looks like an automated wsp trimming ", " vs ","

File: integration/apps/minife/0004-Add-Epoch-markup.patch (hunk 1 above)
Comment (lhlawson) [reply integration/apps/minife/0004-Add-Epoch-markup.patch]: Good catch.  This was a failed vimdiff push of the new header from the 0003 patch to the 0004 patch that included the subject.

File: integration/apps/minife/0005-Add-region-markup.patch (hunk 2 above)
Comment (lhlawson) [reply integration/apps/minife/0005-Add-region-markup.patch]: Failed diff & change push for header.  Fixed in latest.

File: integration/apps/minife/0005-Add-region-markup.patch (hunk 3 above)
Comment (bgeltz) [reply integration/apps/minife/0005-Add-region-markup.patch]: If you want to address wsp issues, please pull that into a standalone patch like I did with nekbone/0001-Fix-whitespace-issues.patch.  I'm also fine if you just want to remove this change.  It has nothing to do with adding region markup.

File: integration/apps/amg/0001-Insert-calls-to-geopm-profile-APIs-in-the-AMG-timing.patch [hunk 4]
```diff
@@ -1,31 +1,19 @@
...
+ 3 files changed, 35 insertions(+), 1 deletion(-)
 
-diff --git a/.gitignore b/.gitignore
-new file mode 100644
-index 0000000..151a620
---- /dev/null
-+++ b/.gitignore
-@@ -0,0 +1,2 @@
-+*.o
-+*.a
```
Comment (bgeltz) [inline integration/apps/amg/0001-Insert-calls-to-geopm-profile-APIs-in-the-AMG-timing.patch]: Why was the .gitignore removed?  This makes the build tree more clean to explore, especially when creating patch files.

File: integration/apps/minife/0005-Add-region-markup.patch [hunk 5]
```diff
@@ -7,31 +7,68 @@ Subject: [PATCH 5/5] Add region markup
...
++void write_matrix(const std::string& filename,
+                   MatrixType& mat)
+ {
+   typedef typename MatrixType::LocalOrdinalType LocalOrdinalType;
+@@ -200,7 +205,7 @@ sum_in_symm_elem_matrix(size_t num,
+   bool flag = false;
+   for(size_t i=0; i<num; ++i) {
+     GlobalOrdinal row = indices[i];
+- 
++
```
Comment (bgeltz) [inline integration/apps/minife/0005-Add-region-markup.patch]: Please pull wsp changes into a separate patch if you're going to address them.

File: integration/apps/pennant/0007-Updated-header-includes-with-geopm_hint.h.patch [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
++#include "geopm_internal.h"
+ #endif
+ #include "PennantGeopm.hh"
+ 
+@@ -580,7 +581,7 @@ void Hydro::sumEnergy(
+         const int slast) {
+ 
+     // compute internal energy
+-    double sumi = 0.; 
++    double sumi = 0.;
```
Comment (bgeltz) [inline integration/apps/pennant/0007-Updated-header-includes-with-geopm_hint.h.patch]: wsp

**Review by bgeltz — CHANGES_REQUESTED** [review]
Please try to isolate the whitespace changes from the required changes for GEOPM.  It's confusing otherwise.

**Review by bgeltz — APPROVED** [review]
